src/main/java/fitr/command/AddGoalCommand.java

```diff
@@ -1,19 +1,95 @@
 package fitr.command;
 
+import fitr.Goal;
 import fitr.list.ExerciseList;
 import fitr.list.FoodList;
+import fitr.list.GoalList;
 import fitr.storage.Storage;
+import fitr.ui.Ui;
 import fitr.user.User;
 
+import java.io.IOException;
+
+import static fitr.common.Commands.COMMAND_FOOD;
+import static fitr.common.Commands.COMMAND_EXERCISE;
+import static fitr.common.Commands.COMMAND_CUSTOM;
+
 public class AddGoalCommand extends Command {
 
     public AddGoalCommand(String command) {
         this.command = command;
     }
 
     @Override
-    public void execute(FoodList foodList, ExerciseList exerciseList, Storage storage, User user) {
-
+    public void execute(FoodList foodList, ExerciseList exerciseList, Storage storage,
+                        User user, GoalList goalList) {
+        Ui.printCustomMessage("----------Goals Section-------------");
+        Ui.printCustomMessage("The following are understood by Fitr:\n"
+                + "'food'      format: food <goal description>\n"
+                + "'exercise'  format: exercise <goal description>\n"
+                + "'custom'    format: custom <goal description>\n"
+                + "'back'      to return back to the main section");
+        String userInput = Ui.read().toLowerCase().trim();
+        while (!userInput.equals("back")) {
+            try {
+                String goalType = userInput.split(" ", 2)[0].trim();
+                switch (goalType) {
+                //Food goal
+                case COMMAND_FOOD:
+                    userInput = userInput.split(" ", 2)[1].trim();
+                    //if (compare(userInput.charAt(0), '>') == 0) {
+                    //    userInput = userInput.replaceFirst("<", "").trim();
+                    //    goalList.addGoal(new Goal(COMMAND_FOOD, userInput));
+                    //} else if (compare(userInput.charAt(0), '<') == 0) {
+                    //    userInput = userInput.replaceFirst(">", "").trim();
+                    //    goalList.addGoal(new Goal(COMMAND_FOOD, userInput));
+                    //} else {
+                    //    goalList.addGoal(new Goal(COMMAND_FOOD, userInput));
+                    //}
```

Comment:
Are these commented lines still needed? If not needed then maybe can consider removing them?